### PR TITLE
TMEDIA-755 simple list block

### PR DESCRIPTION
**Pull request by annasherman**
## Description

Simple List block is migrated to the new architecture.

## Jira Ticket

- [TMEDIA-755](https://arcpublishing.atlassian.net/browse/TMEDIA-755)

## Acceptance Criteria

Simple List block is migrated to the new architecture.

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-755-simple-list-block`
2. Use Storybook and Chromatic to validate changes

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings. -- there are linter errors in unrelated blocks and shared styles...I didn't touch them
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive, please explain why (so that we can fix it whenever it gets refactored).
- [] Confirmed relevant documentation has been updated/added. -- what is the relevant documentation, if any? 

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
The heading has a truncationLines prop that can be used instead of in scss. I remember us talking about this in pairing https://61eeeb27215b5f003acdf18f-ojyukizixo.chromatic.com/?path=/docs/components-heading--heading#properties

Screen Shot 2022-07-26 at 09 29 35

File: .storybook/themes/news.scss [hunk 1]
```diff
@@ -466,6 +465,63 @@
...
+					font-size: var(--global-font-size-7),
+				),
+				simple-list-item: (
+					gap: var(--global-spacing-6),
+				),
+				simple-list-item-anchor: (
+					flex: 0 0 33%,
+				),
+				simple-list-headline-text: (
+					-webkit-line-clamp: 3,
```
Comment (JackHowa) [inline .storybook/themes/news.scss]:
```suggestion
```

can be removed if you add the props `truncationLines` to the Heading

Comment (annasherman) [reply .storybook/themes/news.scss]: powerful info

Comment (annasherman) [reply .storybook/themes/news.scss]: looks like this handles the ellipsis as well so this whole class can go

Comment (JackHowa) [reply .storybook/themes/news.scss]: woo!

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 2]
```diff
@@ -162,13 +152,13 @@ const SimpleList = (props) => {
 									placeholderResizedImageOptions={placeholderResizedImageOptions}
 									targetFallbackImage={targetFallbackImage}
 									arcSite={arcSite}
-									imageProps={imageProps}
+									primaryLogoAlt={primaryLogoAlt}
 								/>
 								<hr />
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx]: This hr can be removed. Instead, the `divider` prop can be used in the stack 

https://61eeeb27215b5f003acdf18f-ojyukizixo.chromatic.com/?path=/story/components-stack--vertical-with-divider

`<Stack divider>`

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 3]
```diff
@@ -142,8 +131,8 @@ const SimpleList = (props) => {
 
 	return (
 		<HeadingSection>
-			<div key={id} className="list-container layout-section">
-				{title ? <Heading className="list-title">{title}</Heading> : null}
+			<Stack key={id} className={BLOCK_CLASS_NAME}>
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx:131]:
```suggestion
			<Stack key={id} className={BLOCK_CLASS_NAME} divider>
```

File: blocks/simple-list-block/package.json [hunk 4]
```diff
@@ -2,15 +2,14 @@
   "name": "@wpmedia/simple-list-block",
   "version": "5.15.0",
   "description": "Simple List – Arc Block",
-  "author": "Jack Howard <jack.howard@washpost.com>",
-  "contributors": [
-    "Beltran Caliz <beltran.caliz@washpost.com>"
-  ],
+  "author": "Arc XP - Themes",
+  "contributors": [],
```
Comment (JackHowa) [inline blocks/simple-list-block/package.json]:
```suggestion
```

small thing -- not a huge deal

File: blocks/simple-list-block/package.json [hunk 5]
```diff
@@ -25,11 +24,7 @@
     "@arc-fusion/prop-types": "^0.1.5",
     "@wpmedia/engine-theme-sdk": "*",
     "@wpmedia/news-theme-css": "*",
```
Comment (JackHowa) [inline blocks/simple-list-block/package.json]: this `news-theme-css` dependency can be removed as it's not used any more

Comment (JackHowa) [reply blocks/simple-list-block/package.json]:
```suggestion
```

File: package.json [hunk 6]
```diff
@@ -120,9 +120,9 @@
     "@wpmedia/author-content-source-block": "file:blocks/author-content-source-block",
     "@wpmedia/byline-block": "file:blocks/byline-block",
     "@wpmedia/card-list-block": "file:blocks/card-list-block",
-    "@wpmedia/content-api-source-block": "file:blocks/content-api-source-block",
     "@wpmedia/collections-content-source-block": "file:blocks/collections-content-source-block",
     "@wpmedia/commerce-product-content-source-block": "file:blocks/commerce-product-content-source-block",
+    "@wpmedia/content-api-source-block": "file:blocks/content-api-source-block",
```
Comment (JackHowa) [inline package.json]: deliberate change here?

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 7]
```diff
@@ -3,12 +3,13 @@ import React from "react";
 import PropTypes from "@arc-fusion/prop-types";
 import { useContent } from "fusion:content";
 import { useFusionContext } from "fusion:context";
-import { LazyLoad, isServerSide } from "@wpmedia/engine-theme-sdk";
-import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
-import { Heading, HeadingSection } from "@wpmedia/shared-styles";
 import getProperties from "fusion:properties";
+import { LazyLoad } from "@wpmedia/engine-theme-sdk";
+import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
+import { Stack, isServerSide, Heading, HeadingSection } from "@wpmedia/arc-themes-components";
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx]: nice work adding isServerSide from the theme components :)

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 8]
```diff
@@ -3,12 +3,13 @@ import React from "react";
 import PropTypes from "@arc-fusion/prop-types";
 import { useContent } from "fusion:content";
 import { useFusionContext } from "fusion:context";
-import { LazyLoad, isServerSide } from "@wpmedia/engine-theme-sdk";
-import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
-import { Heading, HeadingSection } from "@wpmedia/shared-styles";
 import getProperties from "fusion:properties";
+import { LazyLoad } from "@wpmedia/engine-theme-sdk";
+import { extractResizedParams, extractImageFromStory } from "@wpmedia/resizer-image-block";
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx]: these functions (under different names :( ) can be used from the theme components repo. nearly there! 

if you look at the example from Numbered list pr, that might help 🦉  

https://github.com/WPMedia/arc-themes-blocks/pull/1382/files#diff-699df829c47e7e1bdc79e6e42c2892209a9000ac48f8708cb237c0da73d00ebbR10

File: package.json (hunk 6 above)
Comment (annasherman) [reply package.json]: no? that's weird....i didnt think i had touched this file

Comment (annasherman) [reply package.json]: I think by saving in this file, the linter/prettier alphabetized this list

Comment (annasherman) [reply package.json]: "con" comes after "com"

Comment (JackHowa) [reply package.json]: oh interesting. I wonder if you have a global prettier setup

Comment (JackHowa) [reply package.json]: no worries

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 2 above)
Comment (annasherman) [reply blocks/simple-list-block/features/simple-list/default.jsx]: Added!

File: blocks/simple-list-block/package.json (hunk 4 above)
Comment (annasherman) [reply blocks/simple-list-block/package.json]: Fixed

File: blocks/simple-list-block/package.json (hunk 5 above)
Comment (annasherman) [reply blocks/simple-list-block/package.json]: Fixed

Comment (JackHowa) [reply blocks/simple-list-block/package.json]: cool thanks -- sorry for nitpick

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 9]
```diff
@@ -142,33 +131,36 @@ const SimpleList = (props) => {
...
+										websiteDomain={websiteDomain}
+										showHeadline={showHeadline}
+										showImage={showImage}
+										resizedImageOptions={resizedImageOptions}
+										placeholderResizedImageOptions={placeholderResizedImageOptions}
+										targetFallbackImage={targetFallbackImage}
+										arcSite={arcSite}
+										primaryLogoAlt={primaryLogoAlt}
+									/>
+									{/* <hr /> */}
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx]: nice without the hr! would be able to delete the fragment as well and the comment hr now :)

Comment (annasherman) [reply blocks/simple-list-block/features/simple-list/default.jsx]: oooops

File: blocks/simple-list-block/features/simple-list/default.test.jsx [hunk 10]
```diff
@@ -156,6 +156,12 @@ describe("Simple list", () => {
 		expect(wrapper.find("StoryItem").length).toBe(2);
 	});
 
+	it("should render content only for the arcSite", () => {
```
Comment (annasherman) [inline blocks/simple-list-block/features/simple-list/default.test.jsx:159]: @JackHowa how do you feel about this switch? putting this test before the "should not render items when no data provided" seemed to resolve a mock issue. I'm not that great at Jest so happy to take feedback

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 3 above)
Comment (annasherman) [reply blocks/simple-list-block/features/simple-list/default.jsx:131]: sorted!

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 8 above)
Comment (annasherman) [reply blocks/simple-list-block/features/simple-list/default.jsx]: i think i handled this -- took out the placeholderImage stuff as well (per the numbered list PR).

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 3 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.jsx:131]: nice

File: blocks/simple-list-block/features/simple-list/default.jsx [hunk 11]
```diff
@@ -142,33 +128,34 @@ const SimpleList = (props) => {
...
-									imageProps={imageProps}
-								/>
-								<hr />
-							</React.Fragment>
-						))}
+					<Stack className={`${BLOCK_CLASS_NAME}__items`} divider>
+						{contentElements
+							.reduce(unserializeStory(arcSite), [])
+							.map(({ id: listItemId, itemTitle, imageURL, websiteURL, resizedImageOptions }) => (
+								<React.Fragment key={listItemId}>
```
Comment (JackHowa) [inline blocks/simple-list-block/features/simple-list/default.jsx]: oh this fragment can be deleted as well now that we don't have the hr, right?

File: blocks/simple-list-block/features/simple-list/default.test.jsx (hunk 10 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.test.jsx:159]: the order shouldn't matter with mocking. but we can pair -- or you and Joe -- on mocking and jest. there's a few other examples of mocking the content and isServerSide in the numbered list block if you're looking for oher examples

File: .storybook/themes/news.scss [hunk 12]
```diff
@@ -95,8 +95,8 @@
 		"spacing-16": 7rem,
 		// 112px
 		"spacing-17": 7.5rem,
-
 		// 120px
+		//Font Weight
```
Comment (JackHowa) [inline .storybook/themes/news.scss:99]: weird is this new?

Comment (annasherman) [reply .storybook/themes/news.scss:99]: Yeah I added it, there was no comment for it. It was also unclear that the comment about the value was below the value, so adding this helped with formatting

File: .storybook/themes/news.scss [hunk 13]
```diff
@@ -466,6 +465,62 @@
 				),
 			),
 			blocks: (
+				simple-list: (
```
Comment (JackHowa) [inline .storybook/themes/news.scss]: I'm going to quickly reorder this. don't want to give some of my alphabetization advocates on my team any heartattacks

File: .storybook/themes/news.scss (hunk 12 above)
Comment (JackHowa) [reply .storybook/themes/news.scss:99]: this is good. helps to read

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 8 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.jsx]: cool nice

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 2 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.jsx]: nice nice

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 9 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.jsx]: gone now -- nice

File: blocks/simple-list-block/features/simple-list/default.jsx (hunk 11 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.jsx]: I updated

File: .storybook/themes/news.scss (hunk 13 above)
Comment (JackHowa) [reply .storybook/themes/news.scss]: updated and reordered after two tries haha by me

**Comment by JackHowa** [comment]
@annasherman updated the base branch to get the package.json and lockfile so there wasn't that big of a diff

**Review by JackHowa — APPROVED** [review]
Working as expected! Good work @annasherman

File: blocks/simple-list-block/features/simple-list/default.test.jsx (hunk 10 above)
Comment (JackHowa) [reply blocks/simple-list-block/features/simple-list/default.test.jsx:159]: updated!

**Comment by JackHowa** [comment]
@haganka good to look at now. Using body styles for the item text. Using heading level 6 for the top title

**Pull request merged**